src/Stratis.Bitcoin/Utilities/FullNodeExtensions.cs

```diff
@@ -34,21 +34,29 @@ public static class FullNodeExtensions
                             Console.WriteLine(exception.Message);
                         }
                     }
-
+                    
                     done.Wait();
                 };
-#if !NOASSEMBLYCONTEXT
+
                 var assemblyLoadContext = AssemblyLoadContext.GetLoadContext(typeof(FullNode).GetTypeInfo().Assembly);
                 assemblyLoadContext.Unloading += context => shutdown();
-#endif
+
                 Console.CancelKeyPress += (sender, eventArgs) =>
                 {
                     shutdown();
                     // Don't terminate the process immediately, wait for the Main thread to exit gracefully.
                     eventArgs.Cancel = true;
                 };
 
-                await node.RunAsync(cts.Token, "Application started. Press Ctrl+C to shut down.", "Application stopped.").ConfigureAwait(false);
+                try
```

Comment:
If there is a try catch here can we remove the one in program.cs?

Reply:
Ah this is to invoke the `done.Set();`
What about putting a finally here and let the exception in program.cs be the only place to log errors?

```
                 try
                 {
                     await node.RunAsync(cts.Token, "Application started. Press Ctrl+C to shut down.", "Application stopped.").ConfigureAwait(false);
                 }
                 finally
                 {
                     done.Set();
                 }
```